### PR TITLE
Add support to 32 bytes chain work to textual checkpoints format 

### PR DESCRIPTION
**Pull request by nathanieliov**
## Description

- Add use of deserializeCompactV2 to support 32 bytes chain work when reading textual checkpoints format
- Add tests for textual format reading

## How Has This Been Tested?

Unit tests

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [X] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [X] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [X] Tests for the changes have been added (for bug fixes / features)
- [ ] Requires Activation Code (Hard Fork)

### CODE REVIEW

**Review by apancorb — APPROVED** [review]

**Comment by apancorb** [comment]
It seems you might be missing two files here?

```
src/test/resources/co/rsk/bitcoinj/core/checkpointmanagertest/noCheckpoints.checkpoints.txt
src/test/resources/co/rsk/bitcoinj/core/checkpointmanagertest/unsupportedFormat.checkpoints.txt

```

**Comment by nathanieliov** [comment]
> noCheckpoints

They exists, see them [here](https://github.com/rsksmart/bitcoinj/tree/3e6d064c59c77d6e7bd4af3c8a7993e51baa3548/core/src/test/resources/org/bitcoinj/core/checkpointmanagertest) in the test/resource folder

File: core/src/main/java/org/bitcoinj/core/CheckpointManager.java [hunk 1]
```diff
@@ -184,15 +184,18 @@ private Sha256Hash readTextual(InputStream inputStream) throws IOException {
...
             for (int i = 0; i < numCheckpoints; i++) {
                 byte[] bytes = BASE64.decode(reader.readLine());
                 hasher.putBytes(bytes);
-                buffer.position(0);
-                buffer.put(bytes);
-                buffer.position(0);
-                StoredBlock block = StoredBlock.deserializeCompactLegacy(params, buffer);
+                ByteBuffer buffer = ByteBuffer.wrap(bytes);
+                StoredBlock block;
+                if (bytes.length == StoredBlock.COMPACT_SERIALIZED_SIZE){
```
[the remark this answers is not in the document]
Comment (nathanieliov) [reply core/src/main/java/org/bitcoinj/core/CheckpointManager.java]: Updated

**Review by jeremy-then — APPROVED** [review]
LGTM.

**Comment by nathanieliov** [comment]
Please review these changes in PR: https://github.com/rsksmart/bitcoinj/pull/23.

**Pull request closed without merging**